Panic error deploying a stack

**Describe the bug**
When I try to run an `okteto stack deploy` for this repository https://github.com/ifbyol/flask-producer-consumer, I'm getting a panic error 

<img width="1063" alt="Captura de pantalla 2022-03-15 a las 19 40 48" src="https://user-images.githubusercontent.com/3510171/158546520-6551a8ee-bd50-49b6-83af-dc54292ae548.png">

In this case is using the UI, but the same happen when you running it locally

```
okteto stack deploy
 !  'okteto stack deploy' is deprecated in favor of 'okteto deploy', and will be removed in version 2.2.0
 i  Using nacho @ okteto.nacho.dev.okteto.net as context
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x5826493]

goroutine 1 [running]:
github.com/okteto/okteto/pkg/cmd/build.OptsFromManifest({_, _}, _, {{0x0, 0x0, 0x0}, {0x0, 0x0, 0x0}, {0x0, ...}, ...})
	/Users/nacho/repositories/okteto/pkg/cmd/build/build.go:211 +0x73
github.com/okteto/okteto/pkg/cmd/stack.translateBuildImages({0x69c70e8, 0xc00019a000}, 0xc0004e7420, 0xc0004e7420)
	/Users/nacho/repositories/okteto/pkg/cmd/stack/translate.go:133 +0x188
github.com/okteto/okteto/pkg/cmd/stack.translate({0x69c70e8, 0xc00019a000}, 0x203000, 0xc0004e7420)
	/Users/nacho/repositories/okteto/pkg/cmd/stack/translate.go:66 +0x55
github.com/okteto/okteto/pkg/cmd/stack.(*Stack).Deploy(0xc0005cdb48, {0x69c70e8, 0xc00019a000}, 0xc0014574a0, 0x0)
	/Users/nacho/repositories/okteto/pkg/cmd/stack/deploy.go:73 +0x9a
github.com/okteto/okteto/cmd/stack.(*DeployCommand).RunDeploy(0xc0005cdc58, {0x69c70e8, 0xc00019a000}, 0xc0014574a0, 0xc0004e7420)
	/Users/nacho/repositories/okteto/cmd/stack/deploy.go:114 +0x256
github.com/okteto/okteto/cmd/stack.deploy.func1(0xc001310c80, {0x832e4f0, 0x0, 0x0})
	/Users/nacho/repositories/okteto/cmd/stack/deploy.go:69 +0x1c5
github.com/spf13/cobra.(*Command).execute(0xc001310c80, {0x832e4f0, 0x0, 0x0})
	/Users/nacho/go/pkg/mod/github.com/spf13/cobra@v1.3.0/command.go:856 +0x60e
github.com/spf13/cobra.(*Command).ExecuteC(0xc0012faf00)
	/Users/nacho/go/pkg/mod/github.com/spf13/cobra@v1.3.0/command.go:974 +0x3bc
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/nacho/go/pkg/mod/github.com/spf13/cobra@v1.3.0/command.go:902
main.main()
	/Users/nacho/repositories/okteto/main.go:137 +0x9bd
```

Additionally, I don't know why we are sending an `Internal Server Error` stage in this case if we have detected that there was a failed command, so it should be a "controlled" error and only show it for the failed command stage

**To Reproduce**
Steps to reproduce the behavior:

1. Clone https://github.com/ifbyol/flask-producer-consumer
2. execute `okteto stack deploy` using the CLI from master version

**Expected behavior**
I shouldn't get any panic error. If there is something wrong with my manifest, I should get information about what is wrong

**Desktop (please complete the following information):**
N/A

**Additional context**
N/A

--- a/pkg/cmd/stack/translate.go
+++ b/pkg/cmd/stack/translate.go
@@ -130,14 +130,16 @@ func translateBuildImages(ctx context.Context, s *model.Stack, options *StackDep
 
 		if !options.ForceBuild {
 			buildInfo := svcInfo.Build
-			opts := build.OptsFromManifest(svcName, buildInfo, build.BuildOptions{})
-			if _, err := registry.GetImageTagWithDigest(opts.Tag); err != oktetoErrors.ErrNotFound {
-				if svcInfo.Image == "" {
-					svcInfo.Image = opts.Tag
+			if buildInfo != nil {
+				opts := build.OptsFromManifest(svcName, buildInfo, build.BuildOptions{})
+				if _, err := registry.GetImageTagWithDigest(opts.Tag); err != oktetoErrors.ErrNotFound {
+					if svcInfo.Image == "" {
+						svcInfo.Image = opts.Tag
+					}
+					continue
 				}
-				continue
+				oktetoLog.Infof("image '%s' not found, building it", opts.Tag)
 			}
-			oktetoLog.Infof("image '%s' not found, building it", opts.Tag)
 		}
 
 		oktetoLog.SetStage(fmt.Sprintf("Building service %s", svcName))

